anydex/wallet/stellar/xlm_wallet.py

```diff
@@ -0,0 +1,253 @@
+            return fail(RuntimeError(f'Stellar wallet with name {self.wallet_name} already exists'))
+
+        self._logger.info('Creating Stellar wallet with name %s', self.wallet_name)
+        keypair = Keypair.random()
+        self.keypair = keypair
+        self.created = True
+        self.account = Account(self.get_address(), self.get_sequence_number())
+        self.stellar_db.add_secret(self.wallet_name, keypair.secret, keypair.public_key)
+
+        return succeed(None)
+
+    def get_balance(self):
+
+        if not self.created:
+            return succeed({
+                'available': 0,
+                'pending': 0,
+                'currency': self.get_identifier(),
+                'precision': self.precision()
+            })
+        xlm_balance = int(float(self.provider.get_balance(
+            address=self.get_address())) * self.stroop_in_lumen())  # balance is not in smallest denomination
+        pending_outgoing = self.stellar_db.get_outgoing_amount(self.get_address())
+        balance = {
+            'available': xlm_balance - pending_outgoing,
+            'pending': 0,  # transactions are confirmed every 5 secs, so is this worth doing?
+            'currency': self.get_identifier(),
+            'precision': self.precision()
+        }
+        return succeed(balance)
+
+    def get_sequence_number(self):
+        """
+        Use either the database or the api to find the sequence number.
+        If using the datbase fails we then use the provider.
+
+        :return: sequence number of this wallet.
+        """
+        sequence_nr_db = self.stellar_db.get_sequence_number(self.get_address())
+        return sequence_nr_db if sequence_nr_db else self.provider.get_account_sequence(
+            self.get_address())
+
+    async def transfer(self, amount, address, memo_id: int = None, asset='XLM'):
+        """
+        Transfer stellar lumens to the specified address.
+        In the future sending other assets might also be possible.
+
+        Normally a payment operation is used, but if the account is not created
+        then an account create operation will be done.
+
+        if you wish to send all of your balance then a merge account operation is used.
+
+        :param amount: amount of lumens to send, in stroop (0.0000001 XLM)
+        :param address: address to sent lumens to. Should be a normal encoded public key.
+        :param memo_id: memo id for sending lumens to exchanges.
+        :param asset: asset type. only XLM is currently supported.
+        :return: Transaction hash
+        """
+        balance = await self.get_balance()
+        fee = self.provider.get_base_fee()  # fee for one operation
+        if balance['available'] - 1 < int(amount) + fee:  # stellar accounts need to hold a minimum of 1 XLM
+            raise InsufficientFunds('Insufficient funds')
+
+        self._logger.info('Creating Stellar Lumens payment with amount %s to address %s', amount, address)
```

Comment:
Just a reminder, code of _transfer_ method seems to be untested from this point onwards. In other words, case where user has sufficient funds to proceed is not tested.

Reply:
Yeah, your right. I didn't think there was much point since it's mostly interaction with the api.